src/main.js

```diff
@@ -80,6 +80,7 @@ const DiscoveryFeedIface = '\
     <method name="hide"> \
       <arg type="u" direction="in" name="timestamp"/> \
     </method> \
+    <method name="hidden" /> \
```

Comment:
This method name is... very confusing IMHO. You're using it to notify when the window from the SideComponent of the DF is destroyed, but by reading this interface one would easily think this refers to the actual discovery feed being hidden. It's also confusing because you now have `hide` and `hidden` methods, which doesn't really help understanding what's going on here.

It looks to me that what you actually want is a new signal (e.g. `SideComponentHidden`) being emitted by the shell via the object implementing the `org.gnome.Shell` D-Bus interface, probably including some parameter so that you can make sure the signal relates to the DF (and not to any other SideComponent window), connect to that (remote) signal from this eos-discovery-feed and run whatever logic you want to run from there.

I understand, though, we're running against the clock here, so if that's too complex and don't have time for that I suppose we could take this, but in that case I'd beg you rename this method to something like `notifySideComponentDestruction` or the like, something different than `hidden` please :-)

Reply:
I would not make it a shell signal, because it would not need to be caught by any other program than the feed... But yeah, I'd rename the method to something like `notifyHidden` or one of Mario's suggestions.

Reply:
`hiddenAnimationCompleted`, perhaps?

I agree with @cosimoc that we probably don't want a signal here - this is just communication between the shell and the discovery feed.

Reply:
I don't want to bikeshed on the names too much, but I strongly thing that the `notify` prefix should be included in whatever name you use here. Otherwise it sounds to me like an event generated in this side of the world, instead of in the shell which is where the event (you're notifying about) is generated.

My suggestion on the signal comes from the fact that the logic coded in here models a typical Observer pattern where there's a "Subject" (the shell) where an event is originated and one "Observer" which gets notified when such event happens. It just happens to be that there's only 1 observer now, but the logic being implemented is still a classic Subject --- notify() --> Observer case, and so I think a signal is still arguably a fit here.

That said, using a signal is an implementation detail, so I'm also fine if you don't want to make it a signal, as long as you call it `notifySomething`. That's fairly important to me :-)

Reply:
Ah! You addressed this already! Sorry about that then (and thanks for addressing my comment)